### PR TITLE
#14255: Add metrics to track chunk allocations and deallocations

### LINKED ISSUE #14255: Add metric to track chunk allocations
I think it would be nice to have chunk allocations/deallocations metrics so we could detect inefficiencies of memory management strategies by continuous monitoring

### PR DESCRIPTION
**Pull request by raipc**
Motivation:

Added metrics to track chunk allocations and deallocations so allocation rate by the arena can be tracked.

Modification:

Added `numChunkAllocations`, `numChunkDeallocations`, `numActiveChunks` to `PoolArenaMetric`

Result:

Fixes #14255.

### CODE REVIEW

File: buffer/src/main/java/io/netty/buffer/PoolArenaMetric.java [hunk 1]
```diff
@@ -92,6 +92,13 @@ public interface PoolArenaMetric extends SizeClassesMetric {
      */
     long numHugeAllocations();
 
+    /**
+     * Return the number of chunks allocations done via the arena.
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PoolArenaMetric.java:96]: Maybe we should return -1 by default and add to the javadocs that -1 is returned when "unknown" ? Same for all the other methods.

**Comment by normanmaurer** [comment]
@chrisvest PTAL as well

File: buffer/src/main/java/io/netty/buffer/PoolArena.java [hunk 2]
```diff
@@ -268,6 +272,9 @@ void freeChunk(PoolChunk<T> chunk, long handle, int normCapacity, SizeClass size
                 }
             }
             destroyChunk = !chunk.parent.free(chunk, handle, normCapacity, nioBuffer);
+            if (destroyChunk) {
+                ++pooledChunkDeallocations;
```
Comment (chrisvest) [inline buffer/src/main/java/io/netty/buffer/PoolArena.java:276]: We could add a comment that all other `destroyChunk` calls comes from the arena itself being finalized, and so don't need to be counted.

**Review by chrisvest — COMMENTED** [review]
Looks good once the comments are resolved.

**Review by chrisvest — APPROVED** [review]

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
@raipc thanks a lot!

**Pull request merged**